ClinGen/clincoded
# Connects to #1077 kl schema casecontrol group

This PR contains schema related changes for case control, including:
 1. Created new schema file `src/clincoded/schemas/caseControl.json` for case control and related test data in file `src/clincoded/tests/data/inserts/caseControl.json`.
 2. Created new schema file `/src/clincoded/schemas/evidenceScore.json` for score and related test dat in file `/src/clincoded/tests/data/inserts/evidenceScore.json`.
 3. Added 5 new data properties for case cohort and control cohort in schema group in file `src/clincoded/schemas/group.json` and edited related test data in file `src/clincoded/tests/data/inserts/group.json`.
 4. Added new data property "caseControlStudies" in schema annotation in file  `src/clincoded/schemas/annotation.json` and edited test data in file `src/clincoded/tests/data/inserts/annotation.json`.
 5. Edited files `src/clincoded/types/__init__.py` to handle related links so that all case control data above can be embedded in GDM.
 6. Edited file `src/clincoded/loadxl.py` for loading score data and case control data.
 7. Generated new dump files.

**Test**
* Case Control
  - Go to [http://localhost:6543/casecontrol/](http://localhost:6543/casecontrol/) and expect to see:
![screen shot 2016-10-19 at 3 36 28 pm](https://cloud.githubusercontent.com/assets/9206696/19540069/d949910c-9611-11e6-9cb4-cbde147ce1fa.png)
  - Click the uuid and expect to see case control data and `caseCohort`, `controlCohort` and `scores` embedded. For example:

![screen shot 2016-10-21 at 3 04 09 pm](https://cloud.githubusercontent.com/assets/9206696/19613161/b21e0024-979f-11e6-9842-ac6cf5d82078.png)
![screen shot 2016-10-21 at 2 00 00 pm](https://cloud.githubusercontent.com/assets/9206696/19611472/ba294ca0-9796-11e6-84f1-a52d90d34ddc.png)
![screen shot 2016-10-21 at 2 00 55 pm](https://cloud.githubusercontent.com/assets/9206696/19611498/d197c632-9796-11e6-8a3e-d077fffda209.png)

* Embedded in GDM
  - Go to [http://localhost:6543/gdm/9ffd7c1e-16d9-11e5-b283-60f81dc5b05a/?format=json](http://localhost:6543/gdm/9ffd7c1e-16d9-11e5-b283-60f81dc5b05a/?format=json) and expect to see case control data embedded in `caseControlStudies` in one annotation in the GDM. For example:
![screen shot 2016-10-21 at 1 55 55 pm](https://cloud.githubusercontent.com/assets/9206696/19611363/4223bf9c-9796-11e6-8330-623ea3e2a70a.png)

**End of Test**
